app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragmentTest.kt

```diff
@@ -0,0 +1,67 @@
+package com.github.sdp.tarjetakuna.ui.scanner
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.sdp.tarjetakuna.MainActivity
+import com.github.sdp.tarjetakuna.R
+import com.github.sdp.tarjetakuna.utils.PermissionGranting.PermissionGranting.grantPermissions
+import com.github.sdp.tarjetakuna.utils.waitForText
+import org.junit.*
+import org.junit.runner.RunWith
+import org.junit.runners.MethodSorters
+
+@RunWith(AndroidJUnit4::class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+class ScannerFragmentTest {
+
+    @get:Rule
+    val activityScenarioRule = ActivityScenarioRule(MainActivity::class.java)
+
+    @Before
+    fun setUp() {
+        Intents.init()
+        grantPermissions()
+        activityScenarioRule.scenario.onActivity {
+            it.changeFragment(R.id.nav_scanner)
+        }
+    }
+
+    @After
+    fun after() {
+        Intents.release()
+        println("after done")
+    }
+
+    @Test
+    fun test_0_grantPermissions() {
+        assert(true)
+    }
+
+    @Test
+    fun test_1_defaultTextValues() {
```

Comment:
This test seems to run indefinitely on my machine despite the time outs, maybe you should put a global timeout on the test

Reply:
good idea, i put it